icc-x-api/icc-crypto-x-api.ts

```diff
@@ -1342,9 +1352,16 @@ export class IccCryptoXApi {
     return this.hcpartyBaseApi
       .getHealthcareParty(hcpId)
       .then(hcp => {
-        const crt = this.getKeychainInBrowserLocalStorageAsBase64(hcp.id!!)
         const opts = hcp.options || {}
+
+        const crt = this.getKeychainInBrowserLocalStorageAsBase64(hcp.id!!)
         _.set(opts, this.hcpPreferenceKeyEhealthCert, crt)
+
+        const crtValidityDate = this.getKeychainValidityDateInBrowserLocalStorage(hcp.id!!)
+        if (opts[this.hcpPreferenceKeyEhealthCertDate] !== crtValidityDate) {
+          _.set(opts, this.hcpPreferenceKeyEhealthCertDate, crtValidityDate)
+        }
```

Comment:
See comment above: if `opts[this.hcpPreferenceKeyEhealthCertDate]` is `undefined`, it will become the string `"undefined"`. I don't think you want that.

Reply:
@cebamps Noted